src/fsharp/FSharp.Core/string.fs

```diff
@@ -8,13 +8,34 @@ namespace Microsoft.FSharp.Core
     open Microsoft.FSharp.Core.Operators
     open Microsoft.FSharp.Core.Operators.Checked
     open Microsoft.FSharp.Collections
+    open Microsoft.FSharp.Primitives.Basics
 
     [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
     [<RequireQualifiedAccess>]
     module String =
+        [<CompiledName("Length")>]
+        let length (str:string) = if isNull str then 0 else str.Length
+
+        let private concatArray sep (strings: string []) =
+            match length sep with
+            | 0 -> String.Concat strings
+            // following line should be used when this overload becomes part of .NET Standard (it's only in .NET Core)
+            //| 1 -> String.Join(sep.[0], strings, 0, strings.Length)
+            | _ -> String.Join(sep, strings, 0, strings.Length)
```

Comment:
Typically these are done as inner functions instead of private ones. Could you make that change?

Reply:
Sure, done, fixed: https://github.com/dotnet/fsharp/pull/9483/commits/d7c95e00055b35484c7d2ae6f789fd3a5fb0f138